tests/test_raster/test_interpolate.py

```diff
@@ -0,0 +1,568 @@
+                method=method,
+                force_scipy_function="map_coordinates",
+                shift_area_or_point=shift_aop,
+            )
+            raster_points_interpn = raster.interp_points(
+                (points_x_rand, points_y_rand),
+                method=method,
+                force_scipy_function="interpn",
+                shift_area_or_point=shift_aop,
+            )
+
+            # Not exactly equal in floating point precision since changes in Scipy 1.13.0,
+            # see https://github.com/GlacioHack/geoutils/issues/533
+            assert np.allclose(raster_points_mapcoords, raster_points_interpn)
+
+        # Check that, outside the edge, the interpolation fails and returns a NaN
+        index_x_edge_rand = [-0.5, -0.5, -0.5, 25, 25, 49.5, 49.5, 49.5]
+        index_y_edge_rand = [-0.5, 25, 49.5, -0.5, 49.5, -0.5, 25, 49.5]
+
+        points_x_rand, points_y_rand = raster.ij2xy(
+            i=index_x_edge_rand, j=index_y_edge_rand, shift_area_or_point=shift_aop
+        )
+
+        # Nearest doesn't apply, just linear and above
+        for method in ["cubic", "quintic"]:
+            raster_points_mapcoords_edge = raster.interp_points(
+                (points_x_rand, points_y_rand),
+                method=method,
+                force_scipy_function="map_coordinates",
+                shift_area_or_point=shift_aop,
+            )
+            raster_points_interpn_edge = raster.interp_points(
+                (points_x_rand, points_y_rand),
+                method=method,
+                force_scipy_function="interpn",
+                shift_area_or_point=shift_aop,
+            )
+
+            assert all(~np.isfinite(raster_points_mapcoords_edge))
+            assert all(~np.isfinite(raster_points_interpn_edge))
+
+    @pytest.mark.parametrize("example", [landsat_b4_path, aster_dem_path])  # type: ignore
+    @pytest.mark.parametrize(
+        "method", ["nearest", "linear", "cubic", "quintic", "slinear", "pchip", "splinef2d"]
+    )  # type: ignore
+    def test_interp_points__real(
+        self, example: str, method: Literal["nearest", "linear", "cubic", "quintic", "slinear", "pchip", "splinef2d"]
+    ) -> None:
+        """Test interp_points for real data, checking in particular the propagation of nodata."""
+
+        # Check the accuracy of the interpolation at an exact point, and between methods
+
+        # Open and crop for speed
+        r = gu.Raster(example)
+        r = r.crop((r.bounds.left, r.bounds.bottom, r.bounds.left + r.res[0] * 50, r.bounds.bottom + r.res[1] * 50))
+        r.set_area_or_point("Area", shift_area_or_point=False)
+
+        # 1/ Test for an individual point (shape can be tricky in 1 dimension)
+        itest = 10
+        jtest = 10
+        x, y = r.ij2xy(itest, jtest)
+        val = r.interp_points((x, y), method=method, force_scipy_function="map_coordinates")[0]
+        val_img = r.data[itest, jtest]
+        if "nearest" in method or "linear" in method:
```

Comment:
why exclude the others? Also the line could be `if method in ["nearest", "linear"]:`.

Reply:
I added a comment to clarify: it is because "cubic" methods and order high orders do not yield the same value as that of the raster grid at its coordinate, only "nearest" and "linear" do.